cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/vertx/MainVerticle.java

```diff
@@ -0,0 +1,244 @@
+ */
+package com.linkedin.kafka.cruisecontrol.vertx;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.reflect.ClassPath;
+import com.linkedin.kafka.cruisecontrol.async.AsyncKafkaCruiseControl;
+import com.linkedin.kafka.cruisecontrol.vertx.generator.OpenApiRoutePublisher;
+import com.linkedin.kafka.cruisecontrol.vertx.generator.Required;
+import io.swagger.v3.core.util.Json;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.media.Schema;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.ext.web.handler.CorsHandler;
+import io.vertx.ext.web.handler.ErrorHandler;
+import io.vertx.ext.web.handler.SessionHandler;
+import io.vertx.ext.web.handler.StaticHandler;
+import io.vertx.ext.web.sstore.LocalSessionStore;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainVerticle extends AbstractVerticle {
+
+  public static final String APPLICATION_JSON = "application/json";
+  private int _port;
+  private String _host;
+  private HttpServer _server;
+  private SwaggerApi _endPoints;
+  private AsyncKafkaCruiseControl _asynckafkaCruiseControl;
+  private MetricRegistry _dropwizardMetricRegistry;
+
+  public MainVerticle(AsyncKafkaCruiseControl asynckafkaCruiseControl, MetricRegistry dropwizardMetricRegistry, int port, String host) {
+    _port = port;
+    _host = host;
+    _asynckafkaCruiseControl = asynckafkaCruiseControl;
+    _dropwizardMetricRegistry = dropwizardMetricRegistry;
+
+  }
+
+  public HttpServer getServer() {
+    return _server;
+  }
+
+  @Override
+  public void start(Future<Void> startFuture) throws Exception {
+
+    _endPoints = new EndPoints(_asynckafkaCruiseControl, _dropwizardMetricRegistry);
+
+    _server = vertx.createHttpServer(createOptions());
+    _server.requestHandler(configurationRouter()::accept);
+    _server.listen(result -> {
+      if (result.succeeded()) {
+        startFuture.complete();
+      } else {
+        startFuture.fail(result.cause());
+      }
+    });
+  }
+
+  @Override
+  public void stop(Future<Void> future) {
+    _endPoints.destroy();
+    if (_server == null) {
+      future.complete();
+      return;
+    }
+    _server.close(future.completer());
+  }
+
+  private HttpServerOptions createOptions() {
+    HttpServerOptions options = new HttpServerOptions();
+    options.setHost(_host);
+    options.setPort(_port);
+    return options;
+  }
+
+  private Router configurationRouter() {
```

Comment:
`configureRouter` might be a better name

Reply:
Also on another note this might be better to be put in `EndPoints`, logically makes more sense there.